### DISCUSSION TITLE
API Query Builder

### DISCUSSION THREAD

**Discussion by dosandco** [directus/directus#13730] [Feature Requests]
I'm a no-coder, and Directus has opened up an incredible world to me - I cannot thank the makers enough!

I'm using Directus as a back-end to various front-end builders (Bubble, Adalo, Stacker, JetAdmin) - one of the things that I've found quite frustrating about using the API is the trial-and-error nature of trying to get really complex/specific filtering to work.  This is something that the App itself is absolutely brilliant at, so I was wondering whether in the Information tab on the right hand side (under here):

<img width="282" alt="image" src="https://user-images.githubusercontent.com/48129960/171988920-f56fe92d-7120-4d46-8d28-d691c4aaccf8.png">

on a filtered view, there could be another panel showing:

(A) the URL query parameter that provides the current view (with a helpful copy button!):

`?sort=-date_updated&filter[category][_eq]=interesting&limit=2`

I'm assuming this is something that Directus is already doing in the background to show the filtered list.

And, as an extra-special bonus:

(B) the SQL statement that this is performing:

`Select * from public.articles where category = 'interesting'  ORDER BY date_updated DESC LIMIT 2`

For linking Directus with other systems (we use Parabola, in particular), these would be huge time-savers!

And, just being greedy now, (C) a button to "view JSON" that would open another tab just showing the JSON response from the query - I'm sure others who are using this for back-end development would appreciate this as well.

**Comment by sxsws** [comment]
Some nice ideas! (C) would maybe come under an API explorer that some ideas have been thrown around for. I personally don't know where in the list of milestones that would come though.

The latest version of Directus has a new feature called data flows, which may help with your workflows instead of parabola. Also, if you're interested in self-hosting, node-red or n8n are very visual.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
